### ISSUE TITLE
bug(profile): activate project unload lifecycle validator

### ISSUE THREAD

**Issue by ss-o** [z-shell/zsh-lint#181]
## Problem

The configured project documentation says `plugin/project-unload-lifecycle`
matches persistent registrations to an exact `*_plugin_unload` definition
across the complete configured input set.

The validator is not reachable through the CLI:

- `rules.Default()` includes `ProjectUnloadLifecycle`, but unconfigured source
  contexts do not satisfy its configured plugin or Zi annex gate.
- `rules.ForProfile(z-shell/project@1)` selects `configuredProjectRules()`,
  which omits `ProjectUnloadLifecycle`.
- The rule itself and direct analyzer tests exist, so the green suite does not
  prove that the CLI-selected configured profile activates it.

This is present on `main` after the configured-analysis foundation and should
be corrected before that behavior is released or documented as stable.

## Expected behavior

An invocation such as:

```sh
zsh-lint --config zsh-lint.json plugin.plugin.zsh lib/unload.zsh
```

must run the documented cross-file lifecycle validator when the configuration
selects a plugin or Zi annex sourced-library profile.

## Acceptance criteria

- Add `ProjectUnloadLifecycle{}` to the complete configured project rule
  profile, or replace it with an equivalently explicit registered validator.
- Add a CLI-level regression test proving that configured multi-file analysis
  reports a registration with no exact unload function.
- Add the matching passing case where another configured source defines the
  exact unload function.
- Pin rule-profile membership so the validator cannot become unreachable
  without a test failure.
- Reconcile the project-configuration contract and generated reference.
- Run `go test ./...`, `go vet ./...`, the docs-generation check, and the
  configured corpus gate.

## Boundary

This issue corrects the existing configured profile. It does not add automatic
file discovery, infer project type, publish a release, or change severity.

**Comment by ss-o** [comment]
Local implementation is prepared and verified.

Changes:

- add `plugin/project-unload-lifecycle` to the configured `z-shell/project@1` rule profile;
- cover the missing-unload finding through the CLI;
- cover the clean cross-file unload definition case;
- pin the configured profile membership in the rule-set test.

Verification:

- `go test ./...`
- `go vet ./...`
- `git diff --check`

No commit, push, or pull request has been created.
